### ISSUE TITLE
[0.4.3] restrict-plus-operands misses an any-typed map callback index with allowAny:false

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1260]
## Summary

When the receiver of .map is any, its callback index parameter is also implicitly any. ESLint reports index + 1 with allowAny:false, but utlint does not. A number[] receiver is a control: its callback index is a number and both tools correctly accept it.

The original reproduction in #1254 is fixed in 0.4.3. This is an additional case, tracked separately. Verified on Node.js 22.23.1, macOS arm64.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.3 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    },
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true},"include":["example.tsx"]}
```

Run each case separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## Reproduction

```tsx
export function example(items:any){return items.map((value,index)=>index+1);}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.
```

utlint:

```text
(no diagnostics)
```

## Control

```tsx
export function example(items:number[]){return items.map((value,index)=>index+1);}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
(no diagnostics)
```

Only the displayed rule is enabled. Neither invocation has a parsing failure.
